features/admin.branding.v1/components/screen-variation-dropdown.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import React, { FunctionComponent, ReactElement, useEffect, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { DropdownProps, Form, Select } from "semantic-ui-react";
+import useBrandingPreference from "../hooks/use-branding-preference";
+import { PreviewScreenType, PreviewScreenVariationType } from "../models/branding-preferences";
+import { BASE_DISPLAY_VARIATION } from "../models/custom-text-preference";
+
+
+/**
+ * Prop types for the screen variation dropdown component.
+ */
+export interface ScreenVariationDropdownPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Current selected screen type.
+     */
+    selectedScreen: PreviewScreenType;
+    /**
+     * Callback to be called when the screen variation is changed.
+     */
+    onChange: (screenVariation: string) => void;
+    /**
+     * Should the dropdown render as a required field.
+     */
+    required?: boolean;
+}
+
+/**
+ * Dropdown component to select the screen.
+ *
+ * @param props - Props injected to the component.
+ * @returns Screen dropdown component.
+ */
+const ScreenVariationDropdown: FunctionComponent<ScreenVariationDropdownPropsInterface> = (
+    props: ScreenVariationDropdownPropsInterface
+): ReactElement => {
+    const {
+        onChange,
+        selectedScreen,
+        required,
+        ["data-componentid"]: componentId } = props;
+
+    const { getScreenVariations } = useBrandingPreference();
+
+    const { t } = useTranslation();
+
+    const [ screenVariations, setScreenVariations ] = useState<PreviewScreenVariationType[]>();
+
+    const [ selectedScreenVariation, setSelectedScreenVariation ]
+        = useState<PreviewScreenVariationType>();
+
+    useEffect(() => {
```

Comment:
Add doc comments for the funcitons and use effects

Reply:
addressed with https://github.com/wso2/identity-apps/pull/6027/commits/6f514ee32c24be59d56b0f82cd4c13abfa87c27b